python/ray/data/datasource/image_datasource.py

```diff
@@ -132,35 +60,28 @@ def create_reader(
         _check_import(self, module="PIL", package="Pillow")
         _check_import(self, module="pandas", package="pandas")
```

Comment:
this should be pyarrow instead ?

Reply:
That's a good point! I think we could actually remove this check, since with my suggestion of using `DelegatingBlockBuilder` below, the underlying representation should be an implementation detail of the block layer.